src/main/java/racing/game/RacingGame.java

```diff
@@ -0,0 +1,43 @@
+package racing.game;
+
+import racing.game.result.RacingResult;
+
+import java.util.List;
+
+public class RacingGame {
+
+    private final List<Car> cars;
+    private final int numberOfAttempts;
+    private final RacingResult racingResult;
+
+    public RacingGame(List<Car> cars, int numberOfAttempts) {
+        this.cars = cars;
+        this.numberOfAttempts = numberOfAttempts;
+        this.racingResult = new RacingResult(numberOfAttempts);
+    }
+
+
+    public RacingResult run() {
+        runRacing();
+        return racingResult;
+    }
+
+
+    private void runRacing() {
+        for (int i = 0; i < numberOfAttempts; i++) {
+            runAndSaveRound(i);
+        }
+    }
+
+    private void runAndSaveRound(int roundNumber) {
+        for (Car car : cars) {
+            car.move();
+            saveCarRoundResult(car, roundNumber);
+        }
+    }
```

Comment:
자동차가 모두 이동을 마치고 난 이후에 시도 횟수에 대한 결과를 저장해보는 구조는 어떻게 생각하시나요?
그리고 시도 횟수를 전달하지 않고도 동일한 로직을 처리해볼 수 있을 것 같아요.
```suggestion
    private void runAndSaveRound() {
        for (Car car : cars) {
            car.move();
        }
        saveCarRoundResult(cars);
    }
```